src/main/com/fulcrologic/fulcro/algorithms/normalized_state_helpers.cljc

```diff
@@ -0,0 +1,303 @@
+             nested structure. keys is a sequence of keys. Any empty maps that result
+             will not be present in the new structure."
+  [m [k & ks :as keys]]
+  (if ks
+    (if-let [nextmap (get m k)]
+      (let [newmap (dissoc-in nextmap ks)]
+        (if (seq newmap)
+          (assoc m k newmap)
+          (dissoc m k)))
+      m)
+    (dissoc m k)))
+
+
+(defn- normalized-paths
+  "Walks the tree in a depth first manner and returns the normalized possible paths"
+  [m]
+  (letfn [(paths* [ps ks m]
+            (reduce-kv
+              (fn [ps k v]
+                (if (map? v)
+                  (paths* ps (conj ks k) v)
+                  (conj ps (conj ks k))))
+              ps
+              m))]
+    (filter #(< (count %) 4)
+            (paths* () [] m))))
+
+
+(>defn remove-entity*
+  "Remove the given entity at the given ident. Also scans all tables and removes any to-one or to-many idents that are
+  found that match `ident` (removes dangling pointers to the removed entity).
+
+  The optional `cascade` parameter is a set of keywords that represent edges that should cause recursive deletes
+  (i.e. it indicates edge names that *own* something, indicating it is safe to remove those entities as well).
+
+  Returns the new state map with the entity(ies) removed."
+
+  ([state-map ident]
+   [map? eql/ident? => map?]
+   (remove-entity* state-map ident #{}))
+
+  ([state1 ident cascade]
+   [map? eql/ident? (s/coll-of keyword? :kind set?) => map?]
+   (let [tables (keep (fn [k]
+                        (let [candidate (get state1 k)]
+                          (when (and (map? candidate) (every? map? (vals candidate)))
+                            k))) (keys state1))
+
+         non-tables (keep (fn [k]
+                            (let [candidate (get state1 k)]
+                              (when (vector? candidate)
+                                [k])))
+                          (keys state1))
+
+         remove-idents-at-path (fn [state1 path]
+                                 (let [v (clojure.core/get-in state1 path)]
+                                   (cond
+                                     (int? v) state1
+                                     (= v ident) (dissoc-in state1 path)
+                                     (every? eql/ident? v) (merge/remove-ident* state1 ident path)
+                                     :else state1)))
+
+         candidate-paths (fn [state table-name]
+                           (filter (fn [a-path]
+                                     (let [vl (clojure.core/get-in state a-path)]
+                                       (if (coll? vl)
+                                         (or
+                                           (some #{ident} vl)
+                                           (= ident vl))
+                                         (= ident (take 2 a-path)))))
+                                   (normalized-paths state)))
+
+         remove-ident-from-table (fn [state1 table]
```

Comment:
This can't be right, not even using `table`

Reply:
Actually, this is a reducing function and the `table` parameter is passed while computing `state-without-entity` on Line 179

Reply:
Oh, now I got what you meant 😅 

I'm on it.

Reply:
I've done some refactoring and it turns out that `tables/non-tables` were redundant since we already had `normalized-paths`. 

Have pushed the updated version.